pandas/core/util/hashing.py

```diff
@@ -264,7 +264,7 @@ def hash_array(
 
     # First, turn whatever array this is into unsigned 64-bit ints, if we can
     # manage it.
-    elif isinstance(dtype, np.bool):
+    elif isinstance(dtype, bool):
```

Comment:
I am not sure if this can ever be correct:

```
In [18]: isinstance(np.dtype(bool), bool) 
Out[18]: False
```
unless `dtype` is not actually a dtype ...